### PR TITLE
Resolving issue #5675: Array data handling in populateFromArray

### PR DESCRIPTION
**Pull request by jdmcnair**
### Context

As described in issue #5675 using array data as a data source sometimes results in null output from `populateFromArray`, causing changes to be rejected.

### How has this been tested?

Extensive in-browser testing, and new test added to Core_populateFromArray.spec.js

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #5675 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by AMBudnik** [comment]
Thank you for adding a test scenario. I've marked the PR as ready for review.

**Comment by jdmcnair** [comment]
I added another commit for a closely related (array data) issue, along with a distinct unit test.

**Comment by AMBudnik** [comment]
Thank you for sharing @jdmcnair 

Travis usually breaks without any reason but after reloading it 2 times I see that you're adding some extra spaces. Can you check that, please?

**Comment by jdmcnair** [comment]
Thanks for the heads up @AMBudnik. All updated and passing checks now.

**Comment by AMBudnik** [comment]
Great! I'll ask our CTO to schedule a code review 👍

File: test/e2e/Core_populateFromArray.spec.js [hunk 1]
```diff
@@ -91,6 +91,41 @@ describe('Core_populateFromArray', () => {
...
+    populateFromArray(0, 0, [[[]]], 0, 0);
+
+    expect(output).toEqual([[0, 0, ['2011'], []]]);
+  });
+
+  it('should populate value for array data when bound data begins as empty with new row', () => {
+    // Resolving issue #5675: https://github.com/handsontable/handsontable/issues/5675
+    let output = null;
+    const dataArray = []; // arrayOfArrays();
+    // dataArray[0][0] = ['2011'];
```
Comment (jansiegel) [inline test/e2e/Core_populateFromArray.spec.js]: Is this line needed? Also, the `// arrayOfArrays();` the line above seems to be a leftover from the previous test case.

Comment (jdmcnair) [reply test/e2e/Core_populateFromArray.spec.js]: Indeed. These were just junk comments that I had added as I modified another copied test case. I've removed them now.

**Comment by jdmcnair** [comment]
@jansiegel I've removed those junk comments you noted.

**Pull request merged**